test/test_main.py

```diff
@@ -0,0 +1,81 @@
+import sys
+from collections import OrderedDict
+sys.path.append('..')
+sys.path.append('./translator/GCP_cloud_function/cloud_function')
+from unittest.mock import MagicMock, patch, call, Mock
+sys.modules['icone_translator'] =Mock()
+import main
+import icone_translator
+import os
+import json
+
+
+@patch('urllib.request')
+def test_get_ftp_file(request):
+    test_url='fake url'
+    test_output=main.get_ftp_file(test_url)
+    request.urlopen.assert_called_with(test_url)
+    assert True
+
+
+@patch('google.cloud.pubsub_v1.PublisherClient')
+@patch.object(main, 'get_ftp_url')
+@patch.object(main, 'get_ftp_file')
+@patch.object(main, 'parse_xml')
+@patch.object(main, 'get_wzdx_schema')
+def test_translate_newest_icone_to_wzdx(get_wzdx_schema, parse_xml, get_ftp_file, get_ftp_url, pubsub):
+#the intent of this magic mock fuction is that we give a valid input ,that publishes data
+    main.get_ftp_url=MagicMock(return_value='')
+    main.get_ftp_file=MagicMock(return_value='')
+    main.parse_xml=MagicMock(return_value='')
+    main.get_wzdx_schema=MagicMock(return_value='')
+    icone_translator.wzdx_creator= MagicMock(return_value='WZDx')
+    icone_translator.validate_wzdx= MagicMock(return_value=True)
+    os.environ['project_id']='project_id'
+    os.environ['wzdx_topic_id']='wzdx_topic_id'
+    main.translate_newest_icone_to_wzdx(None,None)
+    publisher=pubsub().publish
+    publisher.assert_called_with(pubsub().topic_path('project_id', 'wzdx_topic_id'),str.encode(json.dumps('WZDx', indent=2)),origin='auto_icone_translator_ftp cloud function')
+    assert True
+
+
+@patch.object(main, 'get_ftp_credentials')
+def test_get_ftp_url(ftp_credentials):
```

Comment:
Good first test with this method, but we still need the edge cases. What happens if an environment variable isn't set?

Reply:
added some more edge cases to test_get_ftp_url test function according to your instruction and pushed the updated code to dev.